lstchain/calib/camera/r0.py

```diff
@@ -289,6 +310,22 @@ def interpolate_spikes(self, event, tel_id):
             event.r1.tel[self.tel_id].trigger_type = event.r0.tel[self.tel_id].trigger_type
             event.r1.tel[self.tel_id].trigger_time = event.r0.tel[self.tel_id].trigger_time
 
+
+    def convert_dc_to_pe(self, event, tel_id):
+        if not isinstance(event.r1.tel[tel_id].waveform, np.ndarray):
+            return
+
+        waveforms = event.r1.tel[tel_id].waveform
+        event.mon.tel[tel_id].calibration = self.mon_data.tel[tel_id].calibration
+        event.mon.tel[tel_id].pixel_status = self.mon_data.tel[tel_id].pixel_status
+
+        # subtract the pedestal per sample (should we do it?) and multiply for the calibration coefficients
+        #
+        pedestal_per_sample = self.mon_data.tel[tel_id].calibration.pedestal_per_sample[:, :, np.newaxis]
+        dc_to_pe = self.mon_data.tel[tel_id].calibration.dc_to_pe[:, :, np.newaxis]
+        event.r1.tel[tel_id].waveform = (waveforms - pedestal_per_sample) * dc_to_pe
```

Comment:
If you do:
```
event.r1.tel[tel_id].waveform -= pedestal_per_sample
event.r1.tel[tel_id].waveform *= dc_to_pe
```
this will be much more efficient, as no new arrays will have to be created. (a - b) * c creates 2 intermediate new arrays (for the result of a - c and the result of (a - c) * b